pkg/apis/core/validation/validation.go

```diff
@@ -4043,6 +4043,31 @@ func ValidateService(service *core.Service) field.ErrorList {
 		ports[key] = true
 	}
 
+	// Validate TopologyKeys
+	if len(service.Spec.TopologyKeys) > 0 {
+		topoKeys := make(sets.String)
+		topoPath := specPath.Child("topologyKeys")
+		if len(service.Spec.TopologyKeys) > core.MaxServiceTopologyKeys {
+			allErrs = append(allErrs, field.TooMany(topoPath, len(service.Spec.TopologyKeys), core.MaxServiceTopologyKeys))
+		}
+		for i, key := range service.Spec.TopologyKeys {
+			keyPath := topoPath.Index(i)
+			// "" must be the last value specified
```

Comment:
nit: "" is depending on using `""` or `*` so let us use Any when referring to it.

Reply:
done